### PR TITLE
Metadata description in the docs

### PR DESCRIPTION
**Pull request by znicholls**
# Please confirm that this PR has done the following:

- [x] Documentation Added
- [ ] Description in RELEASE_NOTES.md Added

## Adding to RELEASE_NOTES.md (remove section after adding to RELEASE_NOTES.md)

Please add a single line in the release notes similar to the following:

```
- (#XX)[http://link-to-pr.com] Added feature which does something
```

# Description of PR

Improves the documentation about the metadata table.

### CODE REVIEW

**Comment by danielhuppmann** [comment]
Thanks @znicholls for starting the discussion.

You are correct, `meta` is (so far) only intended for categorization and quantitative indicators at the model/scenario level - primarily due to consideration that at this level, it's fairly obvious we need something. Otherwise, you end up with a table with as many rows as `data` (after all, why have meta-tags only for variables but not years or region or units) and with potentially a lot of duplicate rows.

Can you describe a use case where you would need tagging at some intermediate level, which would not be covered by a proper workflow or provenance tracking (see #287)?

**Comment by znicholls** [comment]
> Otherwise, you end up with a table with as many rows as `data` (after all, why have meta-tags only for variables but not years or region or units) and with potentially a lot of duplicate rows.

That's true, but the duplicats rows are only a problem because of the long form data structure of pyam. Given it's timeseries based, storing things internally in wide format (like what comes out of `.timeseries()`) would save a lot of that duplication would make this meta much more feasible. You can still cast to long when you need to for whatever operation. That's the approach we've taken in [scmdata](https://github.com/lewisjared/scmdata) and it's way quicker for filtering and means you can have a timeseries by timeseries metadata table without it becoming stupidly big.

> Can you describe a use case where you would need tagging at some intermediate level, which would not be covered by a proper workflow or provenance tracking (see #287)?

Adding an extra column, `climate_model` for determining where e.g. temperature output came from. That would solve the problem of the mangled variable names like in the SR1.5 database.

**Comment by danielhuppmann** [comment]
> Given it's timeseries based, storing things internally in wide format would save a lot of that duplication would make this meta much more feasible.

But what would happen if you want to add meta-tags for particular years? For example, keeping track that year `2025` was interpolated during the postprocessing (with a particular method), whereas `2020` and `2030` were reported by the modeling team. I'm hesitant to implement a feature that will support some additional dimensions but not all.

> Adding an extra column, `climate_model` for determining where e.g. temperature output came from. That would solve the problem of the mangled variable names like in the SR1.5 database.

It's already possible to add extra-columns in `data`, e.g., to distinguish between multiple `Temperature|Global Mean` timeseries for each scenario. What additional climate-model-specific information would need to be stored in `meta`?

**Comment by znicholls** [comment]
I think I finally get the data model now. I've re-written what do you think?

> I'm hesitant to implement a feature that will support some additional dimensions but not all.

Yep good call (I had misunderstood how pyam works).

> What additional climate-model-specific information would need to be stored in `meta`?

Yep also good call (again I had misunderstood how pyam works).

**Comment by jkikstra** [comment]
Hi @znicholls and @danielhuppmann great discussion! As a new user, I was already wondering about the designed purpose/use of the `meta` column.

I took the liberty to restructure the paragraphs a bit, but left all information the same. To me, all this info seems very relevant and should be in the docs. 

I'm only a bit torn as to whether the bit about data point-by-data point metadata should be described under the `meta` column, as it actually is used under `data` .

So what about splitting up this docs section into two? A "`meta` column: scenario-model specific metadata" section and a "Data point specific metadata" section? (feel free to change these headers into something you deem more clear.)

File: doc/source/data.rst [hunk 1]
```diff
@@ -96,10 +96,32 @@ two :class:`pandas.DataFrame` instances (read the `docs`_):
...
+The metadata table is not intended for specific metadata per individual data point.
+If you want to have metadata at this level, then you should operate on the :code:`data` attribute of the :class:`pyam.IamDataFrame`.
+For example, you could specify whether a given data point is the result of an interpolation or not.
+
+A word of warning for adding data point-specific metadata: :code:`pyam` drops any data rows which have any :code:`NaN` values.
+Similarly, if rows which do not have assigned values, those rows will be dropped by :code:`pyam` or will cause it to behave unexpectedly.
+Hence, if you're adding metadata to :code:`data`, you need to make sure that you **add it to every single row**.
+This begs the question, why does :code:`pyam` drop any data rows which have any :code:`NaN` values?
+The reason is that pandas does not play nicely with :code:`NaN` in many cases (see e.g. `here <https://stackoverflow.com/a/18431417>`_ and `here <https://stackoverflow.com/a/13606221>`_).
+Hence it is simpler to remove all the :code:`NaN`'s, ensuring that :code:`pyam` has a clean dataset on which to operate.
```
Comment (jkikstra) [inline doc/source/data.rst]: I'd suggest the second paragraph of this in a separate section, and let the first paragraph refer to that section.

**Comment by danielhuppmann** [comment]
thanks @jkikstra, good call to separate some of this into an own `data` section. If you all don't mind, I'd like to do another round of editing/rewriting later today or tomorrow.

**Comment by znicholls** [comment]
> If you all don't mind, I'd like to do another round of editing/rewriting later today or tomorrow.

Works for me I won't touch it until that's done then

**Comment by danielhuppmann** [comment]
@znicholls, I think that this is good to be merged after you looked https://github.com/znicholls/pyam/pull/4

**Pull request merged**